test/sharness/t0130-swarm.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+test_description="Test swarm command"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+test_launch_ipfs_daemon
+
+test_expect_success "'ipfs swarm addrs' succeeds" '
+	ipfs swarm addrs >swarm_addrs_out 2>swarm_addrs_err ||
+	test_fsh cat swarm_addrs_out || test_fsh cat swarm_addrs_err
+'
+
+test_swarm_addrs_output() {
+	cat "$1" |
+	while IFS='' read -r line || [[ -n $line ]]
+	do
+		grep '^Qm[0-9A-Za-z]\{44\} ([1-9][0-9]*)$' <<<"$line" >/dev/null || return
+		addr_count="$(cat <<<"$line" | sed "s/^.*(//" | sed "s/)$//")"
+		for i in $(seq "$addr_count")
+		do
+			IFS='' read -r line || [[ -n $line ]] || return
+			grep $'^\t/[^/]*/[^/]*/[^/]*/[^/]*$' <<<"$line" >/dev/null || return
+		done
+	done
```

Comment:
I agree wholeheartedly with @chriscool 

We should stick to POSIX for shells.

There are some tests where it may become easier to test things with tools written in Go. Go is good enough, multi platform, really easy to work with (tooling), and already required for the project. So perhaps we can:

- default to do everything with `sh`
- for convoluted things where a small go tool can really help, try go. but still compile it down to a unixy tool to be put into a sharness test.
- for now, no other languages-- as it might introduce another dependency, which would be painful.